### PR TITLE
feat(workspace): add option to disable cwd check when resolving root directory

### PR DESCRIPTION
**Pull request by kiancross**
### Why?

When resolving the root directory, the current working directory is checked before doing a top-down search:

https://github.com/neoclide/coc.nvim/blob/f3b4a47780829f663740f1d276879060b45e8923/src/util/fs.ts#L63

In some cases, it is preferable to skip this check and *always* do a top-down search, particularly in some mono-repo setups.

### How?

This PR adds the option `workspace.workspaceFolderCheckCwd`, which defaults to `true`, so the default behaviour is unchanged. When this option is set to `false`, the current working directory is not checked.

### CODE REVIEW

**Comment by kiancross** [comment]
Looks like the checks are failing due to neovim/neovim#13733.

File: data/schema.json [hunk 1]
```diff
@@ -800,6 +800,11 @@
         "type": "string"
       }
     },
+    "workspace.checkCwd": {
```
Comment (chemzqm) [inline data/schema.json:803]: Please use `workspace.workspaceFolderCheckCwd`

**Review by chemzqm — CHANGES_REQUESTED** [review]

File: data/schema.json (hunk 1 above)
Comment (kiancross) [reply data/schema.json]: @chemzqm OK I have made this change.

**Review by chemzqm — APPROVED** [review]

**Comment by kiancross** [comment]
@chemzqm It looks like some of the tests are timing out? They started failing after I merged master into this branch...

**Pull request merged**